torch/csrc/distributed/c10d/CUDASymmetricMemory-inl.h

```diff
@@ -57,16 +69,16 @@ cas(uint32_t* addr, uint32_t compare, uint32_t val) {
 
 __device__ __forceinline__ void trap() {
 #if defined(USE_ROCM)
-  assert(0);
+  abort();
```

Comment:
qq can we use trap or it has to be abort?

Reply:
abort() calls trap() under the covers. However, on ROCm, the trap is handled differently inside hip runtime. It collects a gpu core dump and causes linux kernerl to create a core dump of the host application. That's why, I currently have the timeout UTs skipped for ROCm. The functionality is there (meaning timeout happens correctly and we trap()), but there isn't a nice way to test it as the current executing thread exits.

Reply:
nit: do yo mind adding a brief comment like above in the code?